### DISCUSSION TITLE
Graph preview

### DISCUSSION THREAD

**Discussion by oligamiq** [uiua-lang/uiua#269] [Language Development]
If it's an array-oriented programming language, I think having a graph preview would be good.

**Comment by kaikalii** [comment]
Sorry, what do you mean by "graph preview"?

**Reply by oligamiq** [reply]
Line chart, Bar chart., ...to analyze the trends in the array.
Ex) pie chart show percentage of elements

I don't want to show all elements

**Comment by kaikalii** [comment]
I don't plan on adding this.
Maybe you could do something like [this](https://uiua.org/pad?src=0_3_1__4oyKw5cxMFvijaXimoIxMDBdCi4K4qyaQCDiiaEo4oavOkDilqEp4o2Y4oqaCg==)?

**Reply by sroccaserra** [reply]
Nice! Since it's a classic & inspiring APL example, maybe it could be added to the uiuisms? I could propose a PR with this exact code if you like. It would entail more maintenance for you though.

**Comment by Omnikar** [comment]
As it happens, I have recently created [uiua-plot](https://github.com/Omnikar/uiua-plot), which is a library that you can use to create configurable data plots. It currently supports scatterplots, line charts, bar graphs, and histograms. I might add pie charts at some point if there is enough demand for it.
